endpoints/openrtb2/amp_auction_test.go

```diff
@@ -122,6 +122,248 @@ func TestAMPPageInfo(t *testing.T) {
 	assert.Equal(t, "test.somepage.co.uk", exchange.lastRequest.Site.Domain)
 }
 
+func TestConsentThroughEndpoint(t *testing.T) {
+	// gdpr consent string that will come inside our http.Request query
+	const consentString = "BOa71ZYOa71ZYAbABBENA8-AAAAbN7_______9______9uz_Gv_r_f__33e8_39v_h_7_-___m_-3zV4-_lvR11yPA1OrfIrwFhiAw"
+	const DigiTurstID = "digitrustId"
+
+	// Generate a marshaled openrtb.BidRequest that DOESN'T come with a gdpr consent string
+	fullMarshaledBidRequest, err := getTestBidRequest(false, false, "", DigiTurstID)
+	if err != nil {
+		t.Fatalf("Failed to marshal the complete openrtb.BidRequest object %v", err)
+	}
+
+	stored := map[string]json.RawMessage{
+		"1": json.RawMessage(fullMarshaledBidRequest),
+	}
+
+	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList())
+	exchange := &mockAmpExchange{}
+
+	endpoint, _ := NewAmpEndpoint(
+		exchange,
+		newParamsValidator(t),
+		&mockAmpStoredReqFetcher{stored},
+		empty_fetcher.EmptyFetcher{},
+		&config.Configuration{MaxRequestSize: maxSize},
+		theMetrics,
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		[]byte{},
+		openrtb_ext.BidderMap,
+	)
+	request := httptest.NewRequest("GET", fmt.Sprintf("/openrtb2/auction/amp?tag_id=1&gdpr_consent=%s", consentString), nil)
+	recorder := httptest.NewRecorder()
+	endpoint(recorder, request, nil)
+
+	// Assert our bidRequest was valid
+	if !assert.NotNil(t, exchange.lastRequest, "Endpoint responded with %d: %s", recorder.Code, recorder.Body.String()) {
+		return
+	}
+	// Assert our bidRequest had a valid "User" field
+	if !assert.NotNil(t, exchange.lastRequest.User, "Resulting bid request should have a valid User field after passing consent string through endpoint") {
+		return
+	}
+	// Assert our bidRequest had a valid "User.Ext" field
+	if !assert.NotNil(t, exchange.lastRequest.User.Ext, "Resulting bid request should have a valid Ext field after passing consent string through endpoint") {
+		return
+	}
+
+	// Assert string `consent` is found in the User.Ext at all
+	assert.NotContainsf(t, fullMarshaledBidRequest, "consent:"+consentString, fmt.Sprintf("Expected bid request to contain consent string %s \n", consentString))
```

Comment:
You don't need to use `fmt.Sprintf` here. You can just pass the error message and the `consentString` as arguments to the `NotContains` function and it will automatically format the error message for you by substituting the `consentString` value in place of `%s`

Reply:
I agree `fmt.Sprintf` is not needed, removed